src/hotspot/share/gc/shenandoah/shenandoahFullGC.cpp

```diff
@@ -1478,3 +1482,61 @@ void ShenandoahFullGC::phase4_compact_objects(ShenandoahHeapRegionSet** worker_s
 
   heap->clear_cancelled_gc(true /* clear oom handler */);
 }
+
+void ShenandoahFullGC::phase5_restore_generation_accounts() {
```

Comment:
Maybe call this `phase5_restore_generation_usage` for consistency with `establish_usage`?

Reply:
I made this change, and then @shipilev suggested another change below.